crates/walrus-simtest/tests/simtest.rs

```diff
@@ -406,6 +415,169 @@ mod tests {
+                        event_stream_catchup_min_checkpoint_lag: 0,
+                        ..storage_node_config.event_processor_config
+                    },
+                    ..storage_node_config
+                })),
+                None,
+                walrus_cluster.nodes[5].cancel_token.clone(),
+            )
+            .await
+            .id(),
+        );
+
+        let new_node_starting_epoch =
+            simtest_utils::get_current_epoch_from_node(&walrus_cluster.nodes[5]).await;
+
+        tracing::info!("adding stake to the new node");
+        client_arc
+            .as_ref()
+            .as_ref()
+            .stake_with_node_pool(
+                walrus_cluster.nodes[5]
+                    .storage_node_capability
+                    .as_ref()
+                    .unwrap()
+                    .node_id,
+                test_cluster::FROST_PER_NODE_WEIGHT * 3,
+            )
+            .await
+            .expect("stake with node pool should not fail");
+
+        tracing::info!("waiting for the new node to catch up and recover");
+        tokio::time::sleep(Duration::from_secs(150)).await;
+
+        tracing::info!("checking the cluster's health info");
+        let node_health_info = simtest_utils::get_nodes_health_info(&walrus_cluster.nodes).await;
+        let committees = client_arc
+            .inner
+            .get_latest_committees_in_test()
+            .await
+            .unwrap();
+        let current_committee = committees.current_committee();
+
+        assert!(current_committee.contains(&walrus_cluster.nodes[5].public_key));
+        assert!(node_health_info[5].shard_detail.is_some());
+
+        tracing::info!("checking that shards in the new node matches the shards in the committees");
+        let shards_in_new_node_based_on_committee = committees
+            .current_committee()
+            .shards_for_node_public_key(&walrus_cluster.nodes[5].public_key);
+        let shards_in_new_node_owned = node_health_info[5]
+            .shard_detail
+            .as_ref()
+            .unwrap()
+            .owned
+            .clone();
+        assert_eq!(
+            shards_in_new_node_based_on_committee.len(),
+            shards_in_new_node_owned.len()
+        );
+        for shard in shards_in_new_node_owned {
+            assert!(shards_in_new_node_based_on_committee.contains(&shard.shard));
+        }
+
+        assert_eq!(node_health_info[5].node_status, "Active");
+
+        workload_handle.abort();
+
+        blob_info_consistency_check
+            .check_storage_node_consistency_from_epoch(new_node_starting_epoch);
+
+        assert!(
+            catchup_using_event_blobs_triggered.load(Ordering::SeqCst),
+            "catchup using event blobs should be triggered"
+        );
+        assert!(
+            recovery_with_incomplete_history_triggered.load(Ordering::SeqCst),
+            "recovery with incomplete history should be triggered"
+        );
```

Comment:
@halfprice I wonder if I could/should remove this check? It seems to be a bit tricky to consistently trigger this path because there may be gaps of several epochs in which no event blobs are created.

Reply:
I've created a follow-up issue for this.